tests/acceptance/standard/test_transfer_project.py

```diff
@@ -299,3 +299,93 @@ def test__transfer_without_satisfied_prerequisites(
+        )
+        projects_in_destination_before_transfer = other_group.projects.list()
+
+        config = f"""
+        projects_and_groups:
+          {project_new_path_with_namespace}:
+            project:
+              transfer_from: {project_for_function.path_with_namespace}
+        """
+
+        run_gitlabform(config, "ALL_DEFINED")
+        projects_in_destination_after_transfer = other_group.projects.list()
+
+        assert (
+            len(projects_in_destination_after_transfer)
+            == len(projects_in_destination_before_transfer) + 1
+        )
+
+    def test__transfer_with_target_all(self, project_for_function, group, other_group):
+        project_new_path_with_namespace = (
+            f"{other_group.path}/{project_for_function.name}"
+        )
+        projects_in_destination_before_transfer = other_group.projects.list()
+
+        config = f"""
+        projects_and_groups:
+          {project_new_path_with_namespace}:
+            project:
+              transfer_from: {project_for_function.path_with_namespace}
+        """
+
+        run_gitlabform(config, "ALL")
+        projects_in_destination_after_transfer = other_group.projects.list()
+
+        assert (
+            len(projects_in_destination_after_transfer)
+            == len(projects_in_destination_before_transfer) + 1
+        )
+
+    def test__transfer_with_target_group(
+        self, project_for_function, group, other_group
+    ):
+        project_new_path_with_namespace = (
+            f"{other_group.path}/{project_for_function.name}"
+        )
+        projects_in_destination_before_transfer = other_group.projects.list()
+
+        config = f"""
+        projects_and_groups:
+          {project_new_path_with_namespace}:
+            project:
+              transfer_from: {project_for_function.path_with_namespace}
+        """
+
+        run_gitlabform(config, other_group.path)
+        projects_in_destination_after_transfer = other_group.projects.list()
+
+        assert (
+            len(projects_in_destination_after_transfer)
+            == len(projects_in_destination_before_transfer) + 1
+        )
+
+    def test__transfer_with_target_sub_group(
+        self, project_in_other_subgroup, group, other_subgroup
+    ):
+        project_new_path_with_namespace = (
+            f"{group.path}/{project_in_other_subgroup.name}"
+        )
+        projects_in_destination_before_transfer = group.projects.list()
+
+        config = f"""
+        projects_and_groups:
+          {project_new_path_with_namespace}:
+            project:
+              transfer_from: {project_in_other_subgroup.path_with_namespace}
+        """
+
+        run_gitlabform(config, f"{group.path}/{other_subgroup.path}")
+        projects_in_destination_after_transfer = group.projects.list()
+
+        assert (
+            len(projects_in_destination_after_transfer)
+            == len(projects_in_destination_before_transfer) + 1
+        )
```

Comment:
Trying to understand what this test case is doing. Given we have following:

- top level group `foo`
- a subgroup `foo/bar`
- a project in the subgroup `foo/bar/project-a`

Our config looks like this:

```yaml
projects_and_groups:
  foo/project-a:
    project:
      transfer_from: foo/bar/project-a
```

And we run gitlabform like this: `gitlabform --config <config> foo/bar`. This is where I'm getting confused. Our config doesn't contain any "project" reference from `foo`. We only have `foo/project-a`. So, shouldn't this be run as `gitlabform --config foo`, but this kind of doesn't follow the intention of the test case, I think. Something seems not right. When gitlabform targets a subgroup, it will process projects within that subgroup, right? The `transfer_from` could come from anywhere.

What do you think @long-wan-ep ?

Reply:
Ah I think I misunderstood the intention of targeting a group/subgroup. I thought that if a group/subgroup is targetted, it should process all projects in that group/subgroup + projects being transferred from that group/subgroup. Let me revisit this.

Reply:
Updated the logic for group target so that only projects transferring into the targeted group are processed. Updated the test as well to transfer into a subgroup.

Reply:
Thanks for refactoring the code. I think it makes sense now.